python/repositories.bzl

```diff
@@ -97,18 +97,28 @@ def _python_repository_impl(rctx):
     release_filename = rctx.attr.release_filename
     urls = rctx.attr.urls or [rctx.attr.url]
 
+    if rctx.attr.netrc:
+        netrc = read_netrc(rctx, rctx.attr.netrc)
+    elif "NETRC" in rctx.os.environ:
+        netrc = read_netrc(rctx, rctx.os.environ["NETRC"])
+    else:
+        netrc = read_user_netrc(rctx)
+    auth = use_netrc(netrc, urls, rctx.attr.auth_patterns)
+
```

Comment:
Given that this code is taken from https://github.com/bazelbuild/bazel/blob/1cf392ff3918386858b8c038f82c013b1e04be98/tools/build_defs/repo/http.bzl#L116, could you copy the full `_get_auth` and add a comment where the code was taken from?